src/types.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * Represents a function of an unknown amount of arguments.
+ */
+export type UnknownFunction = (...args: unknown[]) => void;
```

Comment:
Not sure if possible, but I in the past I had issues with `UnknownFunction`. I think we should be using concrete types.

Reply:
I could replace the functions of the constructor (onInit, onDestroy, etc) with concrete types. But for the util function debounce we will need the UnknownFunction. Not sure how to do that otherwise.

Reply:
I think the `UnknownFunction` could be changed to accept a function with any arguments.